client/blocks/daily-post-button/test/index.jsx

```diff
@@ -96,16 +109,19 @@ describe( 'DailyPostButton', () => {
 					markPostSeen={ markPostSeen }
 				/>
 			);
-			return new Promise( ( resolve ) => {
-				dailyPostButton.instance().renderSitesPopover = resolve;
-				dailyPostButton.simulate( 'click', { preventDefault: noop } );
-			} );
+
+			expect( screen.queryByTestId( 'sites-popover' ) ).not.toBeInTheDocument();
+
+			const btn = screen.getByRole( 'button' );
+			await userEvent.click( btn );
+
+			expect( screen.getByTestId( 'sites-popover' ) ).toBeInTheDocument();
```

Comment:
I absolutely love how much more readable and meaningful this is compared to what we had before 👍